### PR TITLE
UI + database refactoring to make things simpler, faster and more future proof

### PR DESCRIPTION
**Pull request by magnificus**

### CODE REVIEW

File: src/PPStorage.tsx [hunk 1]
```diff
@@ -281,119 +275,94 @@ export default class PPStorage {
...
-          });
-
-          PPGraph.currentGraph.id = newId;
-
-          InterfaceController.notifyListeners(ListenEvent.GraphChanged, {
-            newId,
-            name,
-          });
-
-          InterfaceController.showSnackBar('New playground was saved');
```
Comment (fakob) [inline src/PPStorage.tsx:380]: Do you think we can add this again? I found it practical.

File: src/PPStorage.tsx [hunk 2]
```diff
@@ -281,119 +275,94 @@ export default class PPStorage {
...
-    updateGraphSearchItems: any
-  ) {
+  renameGraph(graphId: string, newName = undefined) {
     this.db
       .transaction('rw', this.db.graphs, this.db.settings, async () => {
         await this.db.graphs.where('id').equals(graphId).modify({
           name: newName,
         });
-        setActionObject({ id: graphId, name: newName });
-        updateGraphSearchItems();
```
Comment (fakob) [inline src/PPStorage.tsx:339]: Could you try out renaming. When I tried it I got 2 errors:
* from Dexie
* from the MUI graph dropdown
Screenshot 2023-08-03 at 20 37 20

**Review by fakob — APPROVED** [review]
Thanks. Looks like a good cleanup. 
Noted a small bug with rename.

File: src/PPStorage.tsx (hunk 2 above)
Comment (magnificus) [reply src/PPStorage.tsx:339]: I will rewrite it as well

File: src/PPStorage.tsx (hunk 1 above)
Comment (magnificus) [reply src/PPStorage.tsx:380]: It should now always trigger, tell me if it doesnt

File: src/PPStorage.tsx (hunk 2 above)
Comment (magnificus) [reply src/PPStorage.tsx:339]: looks ait for me now, tell me if it doesnt work

**Pull request merged**